### DISCUSSION TITLE
Supporting multiple GraphQL API endpoints

### DISCUSSION THREAD

**Discussion by nallwhy** [Netflix/dgs-framework#1189] [General]
I want to make multiple GraphQL API endpoints like `/user/graphql`, `/admin/graphql` because the view changes depending on who the target is.

ex) 

```graphql
# for /user/graphql
type User {
  name: String!
}

# for /admin/graphql
type User {
  id: Long!
  name: String!
}

```

I can handle it in resolvers or just making different objects like `User`, `UserForAdmin`, but those are tricky or verbose or easy to make mistakes.

**Comment by nallwhy** [comment]
I've used [absinthe](https://github.com/absinthe-graphql/absinthe), and that was easy to make multiple GraphQL API endpoints.

**Comment by foo4u** [comment]
You could accomplish this using multiple Spring application contexts. Here's a [decent overview][0] of how you could accomplish application context peers.  There may be some caveats with DGS but I haven't tried.

[0]: https://www.baeldung.com/spring-boot-context-hierarchy
